### PR TITLE
Fix building OOP API

### PR DESCRIPTION
**Pull request by TheNormalnij**
1. Make rotation argument optional.
2. Fix OOP methods
```lua
local b = Building(1337, 0, 0, 4)
iprint(b.position)
```

### CODE REVIEW

File: Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.cpp [hunk 1]
```diff
@@ -30,10 +30,10 @@ void CLuaBuildingDefs::AddClass(lua_State* luaVM)
 
     lua_classfunction(luaVM, "create", "createBuilding");
 
-    lua_registerclass(luaVM, "Building");
+    lua_registerclass(luaVM, "Building", "Element");
 }
 
-CClientBuilding* CLuaBuildingDefs::CreateBuilding(lua_State* const luaVM, uint16_t modelId, CVector pos, CVector rot, std::optional<uint8_t> interior)
+CClientBuilding* CLuaBuildingDefs::CreateBuilding(lua_State* const luaVM, uint16_t modelId, CVector pos, std::optional<CVector> rot, std::optional<uint8_t> interior)
```
Comment (FileEX) [inline Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.cpp]: According to the coding guidelines, std::uint16_t, std::uint8_t, etc. should be used.

File: Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.h [hunk 2]
```diff
@@ -19,7 +19,7 @@ class CLuaBuildingDefs : public CLuaDefs
     static void AddClass(lua_State* luaVM);
 
     // Buiding create funcs
-    static CClientBuilding* CreateBuilding(lua_State* const luaVM, uint16_t modelId, CVector pos, CVector rot, std::optional<uint8_t> interior);
+    static CClientBuilding* CreateBuilding(lua_State* const luaVM, uint16_t modelId, CVector pos, std::optional<CVector> rot, std::optional<uint8_t> interior);
```
Comment (FileEX) [inline Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.h]: According to the coding guidelines, std::uint16_t, std::uint8_t, etc. should be used.

File: Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.cpp [hunk 3]
```diff
@@ -48,11 +48,18 @@ CClientBuilding* CLuaBuildingDefs::CreateBuilding(lua_State* const luaVM, uint16
...
 
-    ConvertDegreesToRadians(rot);
+    if (rot.has_value())
+    {
+        ConvertDegreesToRadians(rot.value());
+    }
+    else
+    {
+        rot.emplace(CVector(0, 0, 0));
+    }
```
Comment (FileEX) [inline Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.cpp]: You can remove unnecessary braces to improve readability

**Review by FileEX — CHANGES_REQUESTED** [review]

File: Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.h (hunk 2 above)
Comment (TheNormalnij) [reply Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.h]: oh, this is official now ;(
weird

Comment (FileEX) [reply Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.h]: Why did you mark the conversations as resolved if you didn't make any changes?

Comment (TheNormalnij) [reply Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.h]: My bad. I didn't push changes

**Comment by Fernando-A-Rocha** [comment]
PR should be renamed to Fix building OOP and make rotation arg optional

File: Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.h (hunk 2 above)
Comment (TheNormalnij) [reply Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.h]: yeah, my local branch was conflicting with remote

**Review by tederis — APPROVED** [review]

**Pull request merged**